documentation/docs/protocol_specification/components/congestion_control.md

```diff
@@ -1,87 +1,186 @@
+  to fairly share the available throughput. Without access Mana the network would be subject to Sybil attacks, which
+  would incentivise even honest actors to artificially increase its own number of nodes.
 
 * _Timestamp_. Before scheduling a new message, the scheduler verifies whether the message timestamp is valid or not.
 
-* _Message weight_. Weight of a message is used to priority messages over the others and it is calculated depending on the type of message and of the message length.
+* _Message weight_. Weight of a message is used to priority messages over the others and it is calculated depending
+  on the type of message and of the message length.
 
+### Outbox Buffer Management
 
-### Outbox Management
-
-Once the message has successfully passed the message parser checks and is solid, it is enqueued into the outbox for scheduling. The outbox is logically split into several queues, each one corresponding to a different node issuing messages. In this section, we describe the operations of message enqueuing (and dequeuing) into (from) the outbox.
+Once the message has successfully passed the message parser checks, is solid and booked, it is enqueued into the
+outbox buffer for scheduling. The outbox is logically split into several queues, each one corresponding to a
+different node issuing messages. Total outbox buffer size is limited, but individual queues do not have any size
+limit of their own. In this section, we describe the operations of message enqueuing (and dequeuing) into (from) the
+outbox buffer.
 
 The enqueuing mechanism includes the following components:
 
-*   _Classification_. The mechanism identifies the queue where the message belongs to according to the node ID of the message issuer.
-*   _Message enqueuing_. The message is actually enqueued, queue is sorted by message timestamps in increasing order and counters are updated (e.g., counters for the total number of bytes in the queue).
-*   _Message drop_. In some circumstances, due to network congestion or to ongoing attacks, some messages shall be dropped to guarantee bounded delays and isolate attacker's messages. Specifically, a node shall drop messages in two situations:
-    - since buffers are of a limited size, if the total number of bytes in all queues exceeds a certain threshold, new incoming messages are dropped;
-    - to guarantee the security of the network, if a certain queue exceeds a given threshold, new incoming packets from that specific node ID will be dropped.
+* _Classification_. The mechanism identifies the queue where the message belongs to according to the node ID of
+  the message issuer.
+* _Message enqueuing_. The message is actually enqueued, queue is sorted by message timestamps in increasing order
+  and counters are updated (e.g., counters for the total number of bytes in the queue).
+* _Message skipping_. The message is skipped when they are confirmed while still waiting to be scheduled in the
```

Comment:
What do you mean with "skipped"? It means that, at the moment of scheduling, we check if the message has already been confirmed? Or do we regularly check if messages have been confirmed. This may be an implementation detail though. However, you have to mention what to do when "skipping": do you remove the message from the outbox? Do you keep it there? Else?